presto-main/src/test/java/com/facebook/presto/spiller/GenericPartitioningSpillerTest.java

```diff
@@ -0,0 +1,134 @@
+    private static final LocalPartitionGenerator POSITIVE_AND_NEGATIVE_PARTITIONS_HASH_GENERATOR = new PositiveAndNegativePartitionsHashGenerator(0);
+    private final BlockEncodingSerde blockEncodingSerde = new BlockEncodingManager(new TypeRegistry(ImmutableSet.of(BIGINT, DOUBLE, VARBINARY)));
+    private final BinarySpillerFactory factory = new BinarySpillerFactory(blockEncodingSerde, new FeaturesConfig());
+
+    @Test
+    public void testFileSpiller()
+            throws Exception
+    {
+        try (PartitioningSpiller spiller = factory.createPartitioningSpiller(TYPES, POSITIVE_AND_NEGATIVE_PARTITIONS_HASH_GENERATOR, PARTITIONS_COUNT, ImmutableSet.of())) {
+            testSimpleSpiller(spiller);
+        }
+    }
+
+    private void testSimpleSpiller(PartitioningSpiller spiller)
+            throws ExecutionException, InterruptedException
+    {
+        RowPagesBuilder builder = RowPagesBuilder.rowPagesBuilder(TYPES);
+        builder.addSequencePage(10, 0, 5, 10, 15);
+        builder.addSequencePage(10, -10, -5, 0, 5);
+
+        List<Page> firstSpill = builder.build();
+
+        builder = RowPagesBuilder.rowPagesBuilder(TYPES);
+        builder.addSequencePage(10, 10, 15, 20, 25);
+        builder.addSequencePage(10, 20, 25, 30, 35);
+
+        List<Page> secondSpill = builder.build();
+
+        builder = RowPagesBuilder.rowPagesBuilder(TYPES);
+        builder.addSequencePage(30, 0, 5, 10, 15);
+
+        List<Page> positiveRows = builder.build();
+
+        builder = RowPagesBuilder.rowPagesBuilder(TYPES);
+        builder.addSequencePage(10, -10, -5, 0, 5);
+
+        List<Page> negativeRows = builder.build();
+
+        testSpiller(
+                TYPES,
+                spiller,
+                ImmutableList.of(firstSpill, secondSpill),
+                ImmutableList.of(positiveRows, negativeRows));
+    }
+
+    private void testSpiller(List<Type> types, PartitioningSpiller spiller, List<List<Page>> spills, List<List<Page>> expectedPartitions)
+            throws ExecutionException, InterruptedException
+    {
+        for (List<Page> spill : spills) {
+            for (Page page : spill) {
+                spiller.partitionAndSpill(page).isBlocked().get();
+            }
+        }
+
+        for (int partition = 0; partition < expectedPartitions.size(); partition++) {
+            List<Page> actualSpill = ImmutableList.copyOf(spiller.getSpilledPages(partition));
+            List<Page> expectedSpill = expectedPartitions.get(partition);
+
+            assertEquals(actualSpill.size(), expectedSpill.size());
+            for (int j = 0; j < actualSpill.size(); j++) {
+                assertPageEquals(types, actualSpill.get(j), expectedSpill.get(j));
+            }
+        }
+    }
+
+    private static class PositiveAndNegativePartitionsHashGenerator
+            extends LocalPartitionGenerator
+    {
+        private final int bigintChannel;
+
+        public PositiveAndNegativePartitionsHashGenerator(int bigintChannel)
```

Comment:
Renamed to `hashChannel` but actually it has to be bigint, maybe `bigintHashChannel` would be better name?